### ISSUE TITLE
Rewriting linter-javac in ES(6?)

### ISSUE THREAD

**Issue by florianb** [AtomLinter/linter-javac#76]
I am currently working hard making bugs testable, ironing bumps out and enhance the use of classpaths in general.

It unfortunately seems to be a huge effort to make the linter-javac-plugin :smoking:-testable **without** touching the plugin itself. I currently don't want to add any untested functionality to prevent delivering a failure to our users.

Therefore i file this issue to **collect** ideas and feedback on the rewrite which will be accomplished with the ["Release 2.0.0"-Milestone](https://github.com/AtomLinter/linter-javac/milestones/Release%202.0.0).

## :santa: Wish List

- [ ] Enable full smoke tests. :smoking:
- [ ] Separate concerns (put supportive logic into separate modules). :scissors: 
- [ ] Lower startup-time under 50ms. :racehorse:
- [ ] Lint on edit! :racehorse:
- [ ] Create a Wiki for Contributors. :gift_heart: 

# Help me

- finding ideas
- give feedback on the codebase
- create PR's

**Comment by florianb** [comment]
Proof the implementation of performance tests.

**Comment by Arcanemagus** [comment]
I recently put performance metrics into `linter-jscs` (implemented testing a bug and decided to put it in there for good). You can check out how I did that [here](https://github.com/AtomLinter/linter-jscs/blob/v4.0.2/src/linter-jscs.js#L12-L25) as an inspiration.

**Comment by florianb** [comment]
Great idea - thanks @Arcanemagus. I guess i'll implement something similar.

**Comment by Justsnoopy30** [comment]
Is this project abandoned or forgotten about? Nothing has progressed for 4 years.

**Comment by florianb** [comment]
No. I don't use Atom anymore and i don't program Java currently. So i felt no pressure about it.
Feel free to contribute.
